pkg/k8s/apis/cilium.io/v2alpha1/bgp_advert_types.go

```diff
@@ -92,13 +122,22 @@ type Advertisement struct {
 	Attributes *CiliumBGPAttributes `json:"attributes,omitempty"`
 }
 
+// ServiceAdvertisementOptions defines the configuration for ServiceAdvert type.
+type ServiceAdvertisementOptions struct {
+	// Advertisements is a list of BGP service types which needs to be advertised.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default={LoadBalancerIP}
+	Advertisements []BGPServiceAdvertisementType `json:"advertisements,omitempty"`
```

Comment:
Now I realized we have duplicate name "Advertisements":
- `Spec.Advertisements`
- `Spec.Advertisements.Service.Advertisements`

Not sure how much of an issue it is (probably not a much), it is just that the word Advertisements can now have multiple meanings :)

Reply:
Yeah, this is causing some stuttering. 

```
spec:
  advertisements:
    - advertisementType: "Service"
       service:
         advertisements:       <= This could be renamed to serviceTypes
         - LoadBalancerIP
```

Renaming advertisements to serviceTypes seems logical to me, as it is really what it means.

Reply:
In that case just `types` would be sufficient  - so that it is `service.types`.

However that is an issue with `NodePort` not being an option, while we still plan support announcements for services of type `NodePort`.

Maybe it should be `addresses`? As we are advertising different types of service's addresses. For example, for single service of type `LoadBalancer`, we can advertise 3 addresses:
- its `LoadBalancerIP`
- its `ClusterIP`
- its `ExternalIP`

Reply:
Updated to `Spec.Advertisements.Service.Addresses`